src/App/contexts/taskContext.js

```diff
@@ -20,15 +22,48 @@ export const TaskProvider = (props) => {
     else setIsCompletedAll(true);
   };
 
+  // control loading screen
+  function triggerLoading() {
+    setIsLoading(true);
+    setTimeout(() => {
+      setIsLoading(false);
+    }, 900);
+  }
+
+  // Filter data method
+  const getAll = () => {
+    // get all tasks
+    setFilteredData(taskData);
+    setActiveFilter('ALL');
+  };
+
+  const getActive = () => {
+    // get all active tasks
+    const activeTasks = taskData.filter((task) => task.isCompleted === false);
+    setFilteredData(activeTasks);
+    setActiveFilter('ACTIVE');
+  };
+
+  const getCompleted = () => {
+    // get all completed tasks
+    const completedTasks = taskData.filter((task) => task.isCompleted === true);
+    setFilteredData(completedTasks);
+    setActiveFilter('COMPLETED');
+  };
+
+  // Call API method
   async function fetchData() {
     try {
       const response = await axios.get(`${SERVER_URL}`);
-      setTaskData(
-        sortArray(response.data, {
-          by: 'createdAt',
-          order: 'desc',
-        }),
-      );
+      triggerLoading();
+      const sortedData = sortArray(response.data, {
+        by: 'createdAt',
+        order: 'desc',
+      });
+      setTaskData(sortedData);
+      if (activeFilter === 'ALL') {
+        setFilteredData(sortedData);
+      }
```

Comment:
What happens with filteredData if activeFilter is not equal ALL?

Reply:
I changed the way it work, now it cover all situation